mobile-app/src/app/habits/habiticons.tsx

```diff
@@ -4,67 +4,220 @@ import { Text, View } from "@/src/components/Themed";
 import { router } from "expo-router";
 import { useAtom } from "jotai";
-import { useMemo } from "react";
-import { TouchableOpacity } from "react-native";
+import { FlatList, TouchableOpacity } from "react-native";
 import { IconButton } from "./icon-button";
 import { iconStrNameToTablerIcon } from "./icons";
 
 type HabitIconsTuple = [string, string[]];
 
+const categories: HabitIconsTuple[] = [
+  [
+    "Health & Fitness",
+    [
+      "barbell",
+      "yoga",
+      "walk",
+      "run",
+      "swimming",
+      "bike",
+      "bed",
+      "leaf",
+      "bottle",
+      "salad",
+      "apple",
+      "activity",
+      "heartbeat",
+      "AdjustmentsHeart",
+    ],
+  ],
+  [
+    "Productivity & Work",
+    [
+      "deviceLaptop",
+      "clock",
+      "calendar",
+      "briefcase",
+      "edit",
+      "chartDots3",
+      "mail",
+      "table",
+      "deviceDesktop",
+      "fileText",
+      "boxMultiple",
+      "checklist",
+      "buildingEstate",
+      "moodEdit",
+    ],
+  ],
+  [
+    "Personal Development",
+    [
+      "brain",
+      "bulb",
+      "ladder",
+      "school",
+      "deviceMobileOff",
+      "code",
+      "language",
+      "microscope",
+      "certificate",
+      "books",
+      "listSearch",
+      "jewishStar",
+      "trendingUp",
+      "chartArrowsVertical",
+    ],
+  ],
+  [
+    "Leisure & Recreation",
+    [
+      "brush",
+      "music",
+      "camera",
+      "book",
+      "chess",
+      "headphones",
+      "deviceGamepad2",
+      "movie",
+      "beach",
+      "tent",
+      "fishHook",
+      "chefHat",
+      "rollerSkating",
+      "video",
+      "ballBasketball",
+      "ballVolleyball",
+      "ballTennis",
+      "ballBaseball",
+      "ballAmericanFootball",
+      "ballFootball",
+      "pingPong",
+    ],
+  ],
+  [
+    "Social & Family",
+    [
+      "usersGroup",
+      "users",
+      "userHeart",
+      "moodHeart",
+      "moodCrazyHappy",
+      "heart",
+      "hearts",
+      "heartHandshake",
+      "home",
+      "homeHeart",
+      "messageChatbot",
+      "messages",
+      "worldHeart",
+      "confetti",
+    ],
+  ],
+  [
+    "Mindfullness & Spirituality",
+    [
+      "peace",
+      "buildingChurch",
+      "window",
+      "flower",
+      "michelinStar",
+      "planet",
+      "empathize",
+      "plant2",
+      "plant",
+      "trees",
+      "hourglass",
+      "scale",
+      "yinYang",
+      "moon2",
+    ],
+  ],
+  [
+    "Financial",
+    [
+      "buildingBank",
+      "wallet",
+      "chartLine",
+      "reportMoney",
+      "databaseDollar",
+      "coin",
+      "coins",
+      "pigMoney",
+      "homeStats",
+      "zoomMoney",
+      "devicesDollar",
+      "deviceAnalytics",
+      "shoppingCartDollar",
+      "moodDollar",
+    ],
+  ],
+];
+
 export default function HabitIcons() {
   const [icon, setIcon] = useAtom(tempIconAtom);
 
-  const habitIcons: HabitIconsTuple[] = useMemo(() => {
```

Comment:
make a `habitIcons` where you wrap `categories` in a useMemo and use that instead